testing/trino-testing/src/main/java/io/trino/testing/AbstractDynamicFilteringIntegrationSmokeTest.java

```diff
@@ -0,0 +1,777 @@
+import static io.trino.server.DynamicFilterService.DynamicFiltersStats;
+import static io.trino.spi.predicate.Domain.none;
+import static io.trino.spi.predicate.Domain.singleValue;
+import static io.trino.spi.predicate.Range.range;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.sql.analyzer.FeaturesConfig.JoinDistributionType.BROADCAST;
+import static io.trino.sql.analyzer.FeaturesConfig.JoinDistributionType.PARTITIONED;
+import static io.trino.sql.analyzer.FeaturesConfig.JoinReorderingStrategy.NONE;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Generic test for connectors exercising connector's dynamic filter capabilities.
+ *
+ * @see AbstractDynamicFilteringIntegrationSmokeTest
+ */
+public abstract class AbstractDynamicFilteringIntegrationSmokeTest
+        extends AbstractTestQueries
+{
+    private static final int LINEITEM_COUNT = 60175;
+    private static final int ORDERS_COUNT = 15000;
+    private static final int PART_COUNT = 2000;
+    private static final int CUSTOMER_COUNT = 1500;
+
+    protected boolean supportsNodeLocalDynamicFiltering()
+    {
+        return true;
+    }
+
+    protected boolean supportsCoordinatorDynamicFiltering()
+    {
+        return true;
+    }
+
+    @Test(timeOut = 30_000)
+    public void testJoinWithEmptyBuildSide()
+    {
+        skipTestUnless(supportsCoordinatorDynamicFiltering());
+
+        ResultWithQueryId<MaterializedResult> result = getDistributedQueryRunner().executeWithQueryId(
+                getSession(),
+                "SELECT * FROM partitioned_lineitem JOIN supplier ON partitioned_lineitem.suppkey = supplier.suppkey AND supplier.name = 'abc'");
+        assertEquals(result.getResult().getRowCount(), 0);
+
+        // TODO bring back OperatorStats assertions from https://github.com/trinodb/trino/commit/1feaa0f928a02f577c8ac9ef6cc0c8ec2008a46d
+        // after https://github.com/trinodb/trino/issues/5120 is fixed
+
+        DynamicFiltersStats dynamicFiltersStats = getDynamicFilteringStats(result.getQueryId());
+        assertEquals(dynamicFiltersStats.getTotalDynamicFilters(), 1L);
+        assertEquals(dynamicFiltersStats.getLazyDynamicFilters(), 1L);
+        assertEquals(dynamicFiltersStats.getReplicatedDynamicFilters(), 0L);
+        assertEquals(dynamicFiltersStats.getDynamicFiltersCompleted(), 1L);
+
+        DynamicFilterDomainStats domainStats = getOnlyElement(dynamicFiltersStats.getDynamicFilterDomainStats());
+        assertEquals(domainStats.getSimplifiedDomain(), none(INTEGER).toString(getSession().toConnectorSession()));
+        assertEquals(domainStats.getDiscreteValuesCount(), 0);
+        assertEquals(domainStats.getRangeCount(), 0);
+        assertTrue(domainStats.getCollectionDuration().isPresent());
+    }
+
+    @Test
+    public void testJoinWithEmptyBuildSideWithBroadcastJoin()
```

Comment:
this is a new test case?

Reply:
I moved this from TestHiveDistributedJoinQueries#testJoinWithEmptyBuildSide. Probably I should move all cases from that class. There is just one more case remaining there.